Jenkinsfile

```diff
@@ -63,91 +69,146 @@ pipeline {
+          steps {
+            script {
+              lastStage = env.STAGE_NAME
+              if (params.RELEASE_BUILD || isHotfix) {
+                env.VERSION = (isHotfix) ? env.HOTFIX_NAME : params.RELEASE_NAME
+                env.DOCKER_TAG = dockerize_server.docker_build_specify_tag_and_push(env.VERSION, sbtbuild.getDockerPath(project_wd_server), sbtbuild.getDockerArtifact(project_wd_server), 'none')
+              } else {
+                env.DOCKER_TAG = dockerize_server.docker_build('STAGING', env.GIT_COMMIT, sbtbuild.getDockerPath(project_wd_server), sbtbuild.getDockerArtifact(project_wd_server), 'none')
+              }
+            }
+          }
+        }
+        stage('Docker Build Secondary') {
+          steps {
+            script {
+              lastStage = env.STAGE_NAME
+              // Here's where we're getting the secondary artifact (named
+              // via env.STORE_PG_ARTIFACT) out for dockerizing, per the
+              // comment above.
+              if (params.RELEASE_BUILD || isHotfix) {
+                env.SECONDARY_DOCKER_TAG = dockerize_secondary.docker_build_specify_tag_and_push(env.VERSION, sbtbuild.getDockerPath(project_wd_secondary), env.STORE_PG_ARTIFACT, 'none')
+              } else {
+                env.SECONDARY_DOCKER_TAG = dockerize_secondary.docker_build('STAGING', env.GIT_COMMIT, sbtbuild.getDockerPath(project_wd_secondary), env.STORE_PG_ARTIFACT, 'none')
+              }
+              currentBuild.description = (params.RELEASE_DRY_RUN) ? "${env.DOCKER_TAG} & ${env.SECONDARY_DOCKER_TAG} - DRY RUN" : "${env.DOCKER_TAG} & ${env.SECONDARY_DOCKER_TAG}"
+            }
           }
         }
       }
       post {
         success {
           script {
-            if (params.RELEASE_BUILD && !params.RELEASE_DRY_RUN) {
-              Map buildInfoServer = [
-                "project_id": service_server,
-                "build_id": env.DOCKER_TAG,
-                "release_id": params.RELEASE_NAME,
-                "git_tag": env.GIT_TAG
-              ]
-              createBuild(
-                buildInfoServer,
-                rmsSupportedEnvironment.production
-              )
+            if (isHotfix) {
+              env.GIT_TAG = releaseTag.create(env.HOTFIX_NAME)
+            } else if (params.RELEASE_BUILD && !params.RELEASE_DRY_RUN) {
+              env.GIT_TAG = releaseTag.getFormattedTag(params.RELEASE_NAME)
+              if (releaseTag.doesReleaseTagExist(params.RELEASE_NAME)) {
+                echo "REBUILD: Tag ${env.GIT_TAG} already exists"
+                return
+              }
+              if (params.RELEASE_DRY_RUN) {
+                echo "DRY RUN: Would have created ${env.GIT_TAG} and pushed it to the repo"
+                return
+              }
+              releaseTag.create(params.RELEASE_NAME)
             }
           }
         }
       }
     }
-    stage('Dockerize Secondary') {
+    stage('Publish') {
```

Comment:
I've added a new top level stage and two nested stages for publishing the docker images.  They all share the `when` clause.  Each nested stage has its own `post` stage since RMS create build is specific to the service.